src/client/Microsoft.Identity.Client/TokenCache.ITokenCacheInternal.cs

```diff
@@ -476,11 +476,10 @@ private MsalAccessTokenCacheItem FilterByKeyId(MsalAccessTokenCacheItem item, Au
                 return CacheFallbackOperations.GetAdalEntryForMsal(
                     Logger,
                     LegacyCachePersistence,
-                    metadata.PreferredCache,
                     aliases,
                     requestParams.ClientId,
                     upn,
-                    requestParams.Account.HomeAccountId?.Identifier);
+                    requestParams.Account.HomeAccountId?.ObjectId);
```

Comment:
This was a subtle in an area of the code that wasn;'t tested - how MSAL actually saves the ADAL cache. ADAL cache expects the object ID, MSAL wsas setting the whole "objId.tenantId". 

The code used to work because the original implementation had logic as "if I am given a unique ID, try to filter by it, but if none match, continue with all tokens choose one randomly."

Our cache tests work because they use only 1 token.